hibernate-core/src/main/java/org/hibernate/boot/internal/SessionFactoryOptionsBuilder.java

```diff
@@ -372,6 +377,38 @@ public SessionFactoryOptionsBuilder(StandardServiceRegistry serviceRegistry, Boo
 				CurrentTenantIdentifierResolver.class,
 				configurationSettings.get( MULTI_TENANT_IDENTIFIER_RESOLVER )
 		);
+		if ( this.currentTenantIdentifierResolver == null ) {
+			final BeanContainer beanContainer = Helper.allowExtensionsInCdi( serviceRegistry ) ? serviceRegistry.requireService( ManagedBeanRegistry.class ).getBeanContainer() : null;
+			if (beanContainer != null) {
+				this.currentTenantIdentifierResolver = beanContainer.getBean(
+						CurrentTenantIdentifierResolver.class,
+						new BeanContainer.LifecycleOptions() {
+							@Override
+							public boolean canUseCachedReferences() {
+								return true;
+							}
+
+							@Override
+							public boolean useJpaCompliantCreation() {
+								return false;
+							}
+						},
+						new BeanInstanceProducer() {
+
+							@Override
+							public <B> B produceBeanInstance(Class<B> beanType) {
+								return null;
+							}
+
+							@Override
+							public <B> B produceBeanInstance(String name, Class<B> beanType) {
+								return null;
+							}
+
+						}
+				).getBeanInstance();
+			}
```

Comment:
Please extract out a function for this, the constructor is already waaaaaay too long.

Reply:
@gavinking I suggest to introduce new util methods for `org.hibernate.resource.beans.internal.Helper` and refactor all `BeanContainer` access to use those methods in another PR, WDYT?

```java
	@SuppressWarnings( "unchecked" )
	public static <T> T getBean(ServiceRegistry serviceRegistry, Class<?> beanType, boolean canUseCachedReferences, boolean useJpaCompliantCreation, T fallback) {
		final BeanContainer beanContainer = allowExtensionsInCdi( serviceRegistry ) ? serviceRegistry.requireService( ManagedBeanRegistry.class ).getBeanContainer() : null;
		if ( beanContainer == null ) {
			return null;
		}
		return (T) beanContainer.getBean(
				beanType,
				new BeanContainer.LifecycleOptions() {
					@Override
					public boolean canUseCachedReferences() {
						return canUseCachedReferences;
					}

					@Override
					public boolean useJpaCompliantCreation() {
						return useJpaCompliantCreation;
					}
				},
				new BeanInstanceProducer() {

					@Override
					public <B> B produceBeanInstance(Class<B> beanType) {
						return (B) fallback;
					}

					@Override
					public <B> B produceBeanInstance(String name, Class<B> beanType) {
						throw new UnsupportedOperationException("The method shouldn't be called");
					}

				}
		).getBeanInstance();
	}
```

Reply:
>  I suggest to introduce new util methods for org.hibernate.resource.beans.internal.Helper and refactor all `BeanContainer` access to use those methods

That sounds fine to me, but please do it now, so that it doesn't get forgotten. Thanks.

Reply:
I created https://hibernate.atlassian.net/browse/HHH-18737 just now.

Reply:
OK, great, thanks 🙏